include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1012 @@
+
+    Objects of this type represent a range of
+    characters decoded from a percent encoded
+    string. The object has no ownership of the
+    underlying character buffer.
+
+    These views are references to the encoded
+    strings that allow cheap copies.
+
+    The correctness of the underlying string
+    is assumed. The values returned by
+    `decoded_view::iterator` when the
+    underlying values are an invalid
+    percent-encoded string is undefined behavior.
+
+    The function @ref validate_pct_encoding
+    can be used to validate the underlying string.
+
+*/
+class decoded_view
+{
+    char const* p_{nullptr};
+    std::size_t n_{0};
+    std::size_t dn_{0};
+    bool plus_to_space_{true};
+
+    using traits_type = std::char_traits<char>;
+
+public:
+    /** Type of a decoded character
+    */
+    using value_type = char;
+
+    /// @copydoc value_type
+    using reference = char;
+
+    /// @copydoc value_type
+    using const_reference = char;
+
+    /** The unsigned integer type used to represent size.
+    */
+    using size_type = std::size_t;
+
+    /** The signed integer type used to represent differences.
+    */
+    using difference_type = std::ptrdiff_t;
+
+    /** A read-only bidirectional iterator to the decoded range.
+
+        This is a read-only bidirectional iterator to
+        the decoded characters.
+
+     */
+#ifdef BOOST_URL_DOCS
+    using const_iterator = __see_below__;
+#else
+    class const_iterator;
+#endif
+
+    /// @copydoc const_iterator
+    using iterator = const_iterator;
+
+    /** Constructor
+
+        Default-constructed objects represent
+        the empty string.
+
+     */
+    BOOST_CONSTEXPR
+    decoded_view(
+        bool plus_to_space = true) noexcept
+        : p_(nullptr)
+        , n_(0)
+        , dn_(0)
+        , plus_to_space_(plus_to_space)
+    {}
+
+    /** Constructor
+     */
+    decoded_view(
+        char const* str,
+        bool plus_to_space = true) noexcept
+        : p_(str)
+        , n_(traits_type::length(str))
+        , dn_(pct_decode_bytes_unchecked(string_view(str)))
+        , plus_to_space_(plus_to_space)
+    {}
+
+    /** Constructor
+     */
+    decoded_view(
+        char const* str,
+        size_type len,
+        bool plus_to_space = true) noexcept
+        : p_(str)
+        , n_(len)
+        , dn_(pct_decode_bytes_unchecked(string_view(str, len)))
+        , plus_to_space_(plus_to_space)
+    {}
+
+    /** Constructor
+     */
+#ifndef BOOST_URL_DOCS
+    template <
+        class End,
+        typename std::enable_if<std::is_same<End, char const*>::value, int>::type = 0>
+    BOOST_CXX14_CONSTEXPR
+    decoded_view(
+        char const* first,
+        End last,
+        bool plus_to_space = true) noexcept
+#else
+    BOOST_CXX14_CONSTEXPR
+    decoded_view(
+        char const* first,
+        char const* last,
+        bool plus_to_space = true) noexcept
+#endif
+        : p_(first)
```

Comment:
Do we need this overload? Doesn't construction from `string_view` cover everything?

Reply:
No answer?